### PR TITLE
Fix GitHub action failures

### PR DESCRIPTION
**Pull request by amir-elgayed**
Fix `golangci-lint` GHA issues:
- Generate `mock_delete_lock_command.go`
- Fix `locking_test.go` and `delete_lock_command_test.go` compilation errors due to changing method signatures
- Add error handling for `bucket#Put` method calls
- `gofmted` `server/events/models/models.go` and `pull_closed_executor.go`
- Remove unnecessary `nil` check
- Remove unnecessary `else` statement when `if` block ends with a `return`
- Variable naming: `PullUrl` to `PullURL`
- Assert no errors when calling `TestGetQueueLock#TryLock`

### CODE REVIEW

File: server/controllers/locks_controller.go [hunk 1]
```diff
@@ -84,20 +84,18 @@ func (l *LocksController) GetLock(w http.ResponseWriter, r *http.Request) {
 	queue, _ = l.Locker.GetQueueByLock(lock.Project, lock.Workspace)
 	var lockDetailQueue []templates.QueueItemIndexData
```
Comment (ghaiszaher) [inline server/controllers/locks_controller.go:86]:
```suggestion
 
```

File: server/controllers/locks_controller.go [hunk 2]
```diff
@@ -83,21 +83,19 @@ func (l *LocksController) GetLock(w http.ResponseWriter, r *http.Request) {
...
-				PullNum:       projectLock.Pull.Num,
-				Path:          projectLock.Project.Path,
-				Workspace:     projectLock.Workspace,
-				Time:          projectLock.Time,
-				TimeFormatted: projectLock.Time.Format("02-01-2006 15:04:05"),
-				PullUrl:       projectLock.Pull.URL,
-				Author:        projectLock.Pull.Author,
-			})
-		}
+
```
Comment (ghaiszaher) [inline server/controllers/locks_controller.go:86]:
```suggestion

```

File: server/controllers/templates/web_templates.go [hunk 3]
```diff
@@ -134,10 +134,10 @@ var IndexTemplate = template.Must(template.New("index.html.tmpl").Parse(`
 			{{ range .Queue}}
 				<a title=
 "Author: {{.Author}}
-Url: {{.PullUrl}}
+Url: {{.PullURL}}
```
Comment (ghaiszaher) [inline server/controllers/templates/web_templates.go:137]: Was it really complaining about the `Url`? I thought that's the convention

**Review by ghaiszaher — APPROVED** [review]
❤️

File: server/controllers/templates/web_templates.go (hunk 3 above)
Comment (amir-elgayed) [reply server/controllers/templates/web_templates.go:137]: yes, it did 😄
https://github.com/runatlantis/atlantis/runs/4414323528?check_suite_focus=true#step:3:129

**Comment by amir-elgayed** [comment]
I will cover test fixes in a separate PR, fixing some method call mocking and adjusting assertions where queue functionality intervenes like queueing if you lock twice instead of failing ...

**Comment by ghaiszaher** [comment]
> I will cover test fixes in a separate PR, fixing some method call mocking and adjusting assertions where queue functionality intervenes like queueing if you lock twice instead of failing ...

Thanks! Let's merge this one and check how actions will look like in our PR

**Comment by amir-elgayed** [comment]
> Let's merge this one and check how actions will look like in our PR

In our PR, the linter action has to be approved by a maintainer to run

File: server/core/db/boltdb.go [hunk 4]
```diff
@@ -108,16 +108,15 @@ func (b *BoltDB) TryLock(newLock models.ProjectLock) (bool, models.ProjectLock,
...
-			}
+		}
+		// checking if current lock is with the same PR
+		var currLock models.ProjectLock
+		if err := json.Unmarshal(currLockSerialized, &currLock); err != nil {
+			return errors.Wrap(err, "failed to deserialize current lock")
+		}
+		if currLock.Pull.Num == newLock.Pull.Num {
+			lockAcquired = true
+			return nil
```
Comment (monikma) [inline server/core/db/boltdb.go:119]: @amir-elgayed see https://github.com/runatlantis/atlantis/pull/1923/files#r954309639, seems we can remove double unmarshalling

**Pull request merged**